### DISCUSSION TITLE
Peer-reviewing for code, technical writing and more

### DISCUSSION THREAD

**Discussion by s-leroux** [EddieHubCommunity/support#3938] [Ideas]
Code review is a common practice in our industry. Beyond finding bugs, it's also very effective to increase collaboration between team members. Notably by promoting teaching and sharing knowledge, increasing team cohesion, and providing mentorship for developers novice at some technology.

Beyond code, peer-reviewing is also applicable to technical writing and video recording. Both regarding the content and the style. This is especially true for us non-native English speakers. To give a practical example, I would greatly benefit from someone reviewing a talk I have to pronounce to fix grammar errors and poor accent. In exchange, I could review code or technical writing.

Currently, if you want your work to be reviewed by the community, you either:

* post an issue with the [review] tag on the GitHub support repository, but the engagement here is relatively low, and the outcome is more a generic appreciation than a detailed review
* post in some chatroom¹ of the Discord server. But the messages there are short-lived, and your request is probably lost long before someone might answer it.

**So, I really miss in the EddieHub community a solution to truly support that peer-reviewing process.**

I don't know exactly which form it should take. Should it be a live session with someone? Or an asynchronous process based on GitHub's issues or a shared GoogleDoc document? Should the reviewer and the reviewer register on some app to mix and match them? Should the system be promoted through gamification, with Unicorn Points to earn for each review?

Comments and suggestions are welcome!
:) Sylvain

--
¹ Is this even allowed on the EddieHub Discord server? In which chatroom?

**Comment by EmmaDawsonDev** [comment]
I know in our maintainer discussions we've mentioned having some live sessions in the discord where we do code reviews together. It would be great if we can somehow get that started.

**Comment by eddiejaoude** [comment]
Could people share Pull Requests rather than whole repos - when a repo gets shared, it is a big ask for someone else to review it all, but if a specific piece of code is shared someone can review it and give specific feedback

**Reply by s-leroux** [reply]
It wasn't my idea to ask for reviews on full repositories. To my mind, the scope was a file or a part of a file. Nothing of a larger scale than that.

@eddiejaoude 
> Could people share Pull Requests rather than whole repos - when a repo gets shared, it is a big ask for someone else to review it all

Pull requests show only diffs. It is fragments of codes out-of-context, and the presentation is somewhat hard to understand for people with little experience in that matter (including non-technical people, for blog post reviews for example). Of course, they are links to bring you to the files, but I nevertheless think the entry barrier is too high here.


I had relative success by asking for review not on the full repo, but on a specific file of the repo. You may even reduce the scope by sharing a link on a specific line range.

Example:
https://github.com/EddieHubCommunity/LinkFree/blob/4ab147a08c7111e1cacebae7e947746d5898f92d/src/Components/ImageLoader.js#L10-L17

The scope of the review is highlighted, but the viewer still has context if needed.

**Reply by eddiejaoude** [reply]
I still think PRs do the same thing, it does show me the changes that the author wants me to see but I can always still get the full context of the repo at that PR point in time - which I do a lot of the time, but also I can use Gitpod to even test out the deployed version of the repo from the PR and have a play

I think we both agree that being focused is important, it is just the way it is does we are suggesting a different process
